### PR TITLE
Refresh Breeze documentation

### PR DESCRIPTION
**Pull request by potiuk**
In preparation for switching to the new Breeze, the documentation
of Breeze needs to be reviewed and updated.

This PR updates the docs to be a bit better structured and contain
more relevant information (and removes/replaces some commands that
have been removed/added)

There are suprisingly little changes, actually. Seems that Breeze
matured and stabilized a lot.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
cc: @Bowrna @edithturn @eladkal @xurror

File: BREEZE.rst [hunk 1]
```diff
@@ -366,73 +370,61 @@ Then run the failed command, copy-and-paste the output from your terminal to the
...
+* Join running interactive shell with ``breeze exec`` command
+* Stop running interactive environment with ``breeze stop`` command
+* Restart running interactive environment with ``breeze restart`` command
+* Execute arbitrary command in the test environment with ``breeze shell`` command
+* Execute arbitrary docker-compose command with ``breeze docker-compose`` command
 
-    * Manage KinD Kubernetes cluster and deploy Airflow to KinD cluster ``breeze kind-cluster`` commands
-    * Run Kubernetes tests  specified with ``breeze kind-cluster tests`` command
-    * Enter the interactive kubernetes test environment with ``breeze kind-cluster shell`` command
+Kubernetes tests related.
```
Comment (eladkal) [inline BREEZE.rst]:
```suggestion
Kubernetes tests related:
```

File: BREEZE.rst [hunk 2]
```diff
@@ -366,73 +370,61 @@ Then run the failed command, copy-and-paste the output from your terminal to the
...
+commands to manage the images as described in the user documentation about
+`building the image <https://airflow.apache.org/docs/docker-stack/build.html>`_
 
-Build documentation:
+Maintainer tasks
+----------------
 
-    * Build documentation with ``breeze build-docs`` command
+Maintainers also can use Breeze for other purposes (those are commands that regular contributors likely
+do not need:
```
Comment (eladkal) [inline BREEZE.rst]:
```suggestion
Maintainers also can use Breeze for other purposes (those are commands that regular contributors likely
do not need):
```

**Review by eladkal — APPROVED** [review]
:tada:

**Pull request merged**